lib/adminLogs.ts

```diff
@@ -0,0 +1,126 @@
+import dbConnector from "@lib/integration/dbConnector";
+import executeQuery from "@lib/integration/queryManager";
+import { logMessage } from "@lib/logger";
+
+enum Action {
+  Create = "Create",
+  Read = "Read",
+  Update = "Update",
+  Delete = "Delete",
+}
+
+/**
+ * Logs usage of privileged functions by admins
+ * @param adminSession
+ * @param action
+ * @param event
+ * @param description
+ */
+const logAdminActivity = async (
+  userId: string,
+  action: Action,
+  event: string,
+  description: string
+): Promise<void> => {
+  try {
+    await executeQuery(
+      await dbConnector(),
+      "INSERT INTO admin_logs (user_id, action, event, description) VALUES ($1, $2, $3, $4)",
+      [userId, action.toString(), event, description]
+    );
+  } catch (error) {
+    logMessage.error(error as Error);
+  }
+};
+
+export const logAdminRoleUpdate = async (
```

Comment:
I'm not sure this is a good way to go in terms of calling logAdminActivity. All these functions are really doing is calling another function. This means that a new function is going to be added every time a new event is to be added. If there is no other logic it would be better to just create a string enum with the events and have logAdminActivity called directly from the main code. Further evidence of this is that these functions are `async` when all they are doing is returning the result of another function. These functions don't need to be async at all.